ndn_python_repo/handle/write_command_handle.py

```diff
@@ -158,4 +166,97 @@ async def fetch_segmented_data(self, name, start_block_id: int, end_block_id: Op
             self.storage.put_data_packet(data_name, data_bytes)
             block_id += 1
         insert_num = block_id - start_block_id
-        return insert_num
\ No newline at end of file
+        return insert_num
+
+    async def fetch_listen(self):
+        """
+        Starts listening for /prefix/fetch_map interests and sends insertion data to
+        the catalog when requests is received.
+        """
+        name = self.prefix + ["fetch_map"]
+        logging.debug("Listening: {}".format(Name.to_str(name)))
+        self.app.route(name)(self._on_interest)
+
+    async def check_insert(self, data_name):
+        """
+        Sends an interest to the catalog and waits for acknowledgement which is basically an
+        empty data packet. Once it gets an acknowledgement it knows that the catalog received the
+        request. The module then does nothing and waits for data request from the catalog. Once
+        the request is received the client responds with a list of insertions and deletions.
+        :param catalog_name: the name of the catalog to which to send the insertion request.
+        """
+        method = 'insert'
+        cmd_param = CatalogCommandParameter()
+        cmd_param.name = self.prefix
+        cmd_param.data_name = data_name
+        cmd_param_bytes = cmd_param.encode()
+
+        name = Name.from_str(self.catalog)
+        name += [method]
+        self.nonce = gen_nonce()
+        name += [str(self.nonce)]
+        logging.info("Name: {}".format(Name.to_str(name)))
+        try:
+            aio.ensure_future(self.send_interest(name, cmd_param_bytes))
```

Comment:
`aio.ensure_future()` returns immediately, so you won't catch any Nack or Timeout exceptions.
You can simply use `await self.send_interest(name, cmd_param_bytes)` here.